typescript-requiring-type-checking.js

```diff
@@ -104,6 +106,8 @@ module.exports = {
 
         '@typescript-eslint/prefer-reduce-type-parameter': 'error',
 
+        '@typescript-eslint/prefer-return-this-type': 'error',
```

Comment:
new rule from update [@typescript-eslint/prefer-return-this-type](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-return-this-type.md)

Reply:
Interesting, I didn't know about the polymorphic 'this' type.